libs/unity/library/Runtime/Scripts/PeerConnection.cs

```diff
@@ -558,75 +544,57 @@ public async Task HandleConnectionMessageAsync(SdpMessage message)
+            OnShutdown.Invoke();
+
+            // Prevent publicly accessing the native peer after it has been deinitialized.
+            // This does not prevent systems caching a reference from accessing it, but it
+            // is their responsibility to check that the peer is initialized.
+            Peer = null;
+
+            // Detach all transceivers. This prevents senders/receivers from trying to access
+            // them during their clean-up sequence, as transceivers are about to be destroyed
+            // by the native implementation.
+            foreach (var mediaLine in _mediaLines)
             {
-                // Fire signals before doing anything else to allow listeners to clean-up,
-                // including un-registering any callback and remove any track from the connection.
-                OnShutdown.Invoke();
-
-                // Prevent publicly accessing the native peer after it has been deinitialized.
-                // This does not prevent systems caching a reference from accessing it, but it
-                // is their responsibility to check that the peer is initialized.
-                Peer = null;
-
-                // Detach all transceivers. This prevents senders/receivers from trying to access
-                // them during their clean-up sequence, as transceivers are about to be destroyed
-                // by the native implementation.
-                foreach (var mediaLine in _mediaLines)
-                {
-                    mediaLine.UnpairTransceiver();
-                }
-
-                // Close the connection and release native resources.
-                _nativePeer.Dispose();
-                _nativePeer = null;
+                mediaLine.UnpairTransceiver();
             }
+
+            // Close the connection and release native resources.
+            _nativePeer.Dispose();
+            _nativePeer = null;
         }
 
         #endregion
 
 
         #region Unity MonoBehaviour methods
 
-        protected override void Awake()
-        {
-            base.Awake();
-
-            // Check in case InitializeAsync() was called first.
-            if (_nativePeer == null)
-            {
-                CreateNativePeerConnection();
-            }
-        }
-
         /// <summary>
-        /// Unity Engine Start() hook
+        /// Unity Engine OnEnable() hook
         /// </summary>
         /// <remarks>
-        /// See <see href="https://docs.unity3d.com/ScriptReference/MonoBehaviour.Start.html"/>
+        /// See <see href="https://docs.unity3d.com/ScriptReference/MonoBehaviour.OnEnable.html"/>
         /// </remarks>
-        private void Start()
+        private void OnEnable()
         {
             if (AutoLogErrorsToUnityConsole)
             {
                 OnError.AddListener(OnError_Listener);
             }
-
-            if (AutoInitializeOnStart)
-            {
-                InitializeAsync();
-            }
+            InitializeAsync();
```

Comment:
I didn't test it but is there any incentive to awaiting here? Is Unity changing the `enabled` state before or after the `OnEnable()` was called?

Reply:
The docs say it's called "just after the object is enabled" so I'd expect after. awaiting here seems pointless - if this becomes async there is probably nothing subscribing to the returned Task in the caller - and waiting seems harmful. So I guess users should not rely on `enabled` implying `Peer != null` and should wait for `OnInitialize`